doc/man7/provider.pod

```diff
@@ -224,6 +216,28 @@ In this case an algorithm implementation is implicitly fetched using
 default search criteria and an algorithm name that is consistent with
 the type of EVP_PKEY being used.
 
+=head3 Algorithm naming
```

Comment:
This is not the best place for this description, as providers are by design capable of supplying more than just cryptographic algorithms.  For example, `OSSL_STORE` implementations are named by URI scheme.

I would suggest a separate file for this.  Still in man7/...  or possibly in internal/man7/, depending on if you think this is public or internal info.

Reply:
I'm not sure why you think this isn't a good place for it. I understand that an identifier in a fetch can be something other than an algorithm name. The page is quite explicit on that point:

````
This is most commonly an algorithm name (this is the case for all EVP
methods), but may also be called something else.
````

But, I don't see why that prevents us from talking about algorithm names here (especially as this is "most commonly" the case). I chose the title "algorithm naming" on purpose because of this issue, rather than "identifier naming".